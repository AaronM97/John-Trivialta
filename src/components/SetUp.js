import React, { Component } from 'react'
import { Link } from 'react-router-dom';

export default class SetUp extends Component {


    state = {
        categories: null,
        difficulty: null
    }

    getCategories = () => {
        fetch("https://opentdb.com/api_category.php")
            .then(response => response.json())
            // .then(data => console.log(data))
            .then(json => this.setState({categories: json.trivia_categories}))
    }

    componentDidMount () {
        this.getCategories();
    }


    render() {

        return (
            <div className="setup">
                <h3>Set up your quiz</h3>
                <form onSubmit={this.props.handleSubmit}>
                    <h5>What category would you like?</h5>
                    <select onChange={this.props.handleCategory}>
                        {/* ternary operator to make sure that the mapping happens after setting state to avoid error of null */}
                        {this.state.categories !== null ? this.state.categories.map(category => (
                            <option key={category.id} value={category.name}>{category.name}</option>
                        )) : null}
                    </select>
                    <br/>
                    <h5>What difficulty would you like?</h5>
                    <select id="difficulty" onChange={this.props.handleDifficulty}>
                        <option value="Easy" defaultValue="selected">Easy</option>
                        <option value="Medium">Medium</option>
                        <option value="Hard">Hard</option>
                    </select>
                    <Link to="/questions" type="submit">Submit</Link>
                </form>
            </div>
        )
    }
}
