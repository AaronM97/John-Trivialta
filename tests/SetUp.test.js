import React from 'react';
import ReactDOM from 'react-dom';
import SetUp from '../src/components/SetUp';
import Enzyme, { shallow, mount } from 'enzyme';

describe('renders Setup without crashing', () => {
    it('smoke test', () => {
        const div = document.createElement('div');
        ReactDOM.render(<SetUp />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('should render correctly', () => {
        expect(shallow(<SetUp />)).toMatchSnapshot();
    })

    it('renders John Trivialta heading', () => {
        const wrapper = shallow(<SetUp />);
        let title = wrapper.find('h3');
        expect(title.text()).toEqual('Set up your quiz');
    });

    it('renders the form', () => {
        const wrapper = shallow(<SetUp />);
        let form = wrapper.find('form');
        expect(form.length).toEqual(1);
    });

    it('renders the Submit link', () => {
        const wrapper = shallow(<SetUp />);
        let link = wrapper.find('a');
        expect(link.length).toEqual(1);
    })

})
