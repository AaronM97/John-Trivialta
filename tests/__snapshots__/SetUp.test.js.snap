// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Setup without crashing should render correctly 1`] = `
<div
  className="setup"
>
  <h3>
    Set up your quiz
  </h3>
  <form>
    <h5>
      What category would you like?
    </h5>
    <select />
    <br />
    <h5>
      What difficulty would you like?
    </h5>
    <select
      id="difficulty"
    >
      <option
        defaultValue="selected"
        value="easy"
      >
        Easy
      </option>
      <option
        value="medium"
      >
        Medium
      </option>
      <option
        value="hard"
      >
        Hard
      </option>
    </select>
    <br />
    <br />
    <a
      href="/questions"
      type="submit"
    >
      Submit
    </a>
  </form>
</div>
`;
